MD-7: Suzuka MEV (#7)
movementlabsxyz/MIP: MD/md-6/README.md

# MD-6: Suzuka Block Validation
- **Description**: Provide means to validate blocks to prevent public DA DOS attacks.
- **Authors**: [Liam Monninger](mailto:liam@movementlabs.xyz)

## Overview

> The current implementation of the Suzuka Network relies on a public and permissionless DA. The blocks are further not validated besides their correctness w.r.t. the Aptos executor. 

Author: Yes, this should be a distinction we make at this point.

> The current implementation of the Suzuka Network relies on a public and permissionless DA. The blocks are further not validated besides their correctness w.r.t. the Aptos executor. 

Reviewer: ```suggestion
The current implementation of the Suzuka Network relies on a public and permissionless DA. Anyone can submit batches to the DA. Blocks, that are created from these batches, are not further validated besides their correctness w.r.t. the Aptos executor. 
```

  5. Recommendations (optional): Suggestions or guidance related to the desideratum.

  Format as:

  ### Desideratum Title

  **User Journey**: [user] can [action].

  **Description**: <More detailed explanation if needed (optional)>

  **Justification**: <Why this is a significant or required desideratum>

  **Recommendations**: <Any specific guidance or suggestions (optional)>

  TODO: Remove this comment before finalizing.
-->
### D1: Model Transaction Fairness in Suzuka Network
**User Journey**: A researcher or protocol implementer can understand what transaction fairness means in the Suzuka Network.

**Justification**: Transaction fairness is a key component of any blockchain network. It is important to understand what it means in the context of the Suzuka Network.

**Recommendations**:
- The strongest form of formally defined fairness is currently [batch-order-fairness](https://link.springer.com/chapter/10.1007/978-3-030-56877-1_16). What would it mean for the Suzuka Network to achieve this. 
- Transaction fairness should consider:
    - Time in flight
    - Congestion
    - Ingress node
    - Causality

### D2: Accept Only Blocks Signed by Staked Validators

> **User Journey**: A staked Suzuka Validator is the only entity that can submit blocks to the network which will later be processed by the Aptos executor.

Reviewer: ```suggestion
**User Journey**: A proposer submits batches to the DA which will later be processed by the Aptos executor. A proposer must be a Suzuka validator.
```

Reviewer: is this the correct picture?

**Justification**: This will prevent DOS attacks on the network by requiring a stake to submit blocks.

**Recommendations**:
- Be sure to consider what is lost in this approach. For example, currently, a developer could bypass a Suzuka Full Node an submit directly to the DA--allowing them to avoid any potential censorship. This would be lost in a naive implementation of this approach, as they would be required to either (a) submit via a full-node or (b) stake themselves.

### D3: Invalidation or Validator-slashing for Producing Unfair Blocks

> **User Journey**: Validators can be slashed for producing unfair blocks.

Reviewer: I think you mean to say proposer. The validators could be the proposer, but this is a different topic. (D2)

Author: Yes, that was the intent.

**Justification**: This will disincentivize validators from producing unfair blocks.

**Recommendations**:
- This would be a highly experimental feature and should be researched and implemented with caution. 

> - Deterministic transaction and block ordering should cover most of the cases of unfairness. These are discussed in [MD-7](../md-7/README.md). However, in theory you could also slash on conditions like including too many transactions from one user, sending too many blocks at a given time, sending too few blocks at a given time, sending blocks of unequal size when transactions are available, etc.

Reviewer: I would argue it only covers a sub category rather than the majority. 

> However, in theory you could also slash on conditions like ...

i am not sure why we need to slash. could we have a rule on the executor side whether to accept a batch or not based on these rules instead? or does the DA remove the batch wrapping? where would the slashing be enforced?

Author: In the case you have the staking requirement mentioned above, would always slash in the settlement contract.
